packages/validator-ajv8/src/validator.ts

```diff
@@ -202,14 +203,45 @@ export default class AJV8Validator<
     errors: ErrorObject[] = []
   ): RJSFValidationError[] {
     return errors.map((e: ErrorObject) => {
-      const { instancePath, keyword, message, params, schemaPath } = e;
+      const {
+        instancePath,
+        keyword,
+        params,
+        schemaPath,
+        parentSchema,
+        ...rest
+      } = e;
+      let { message } = rest;
+
       let property = instancePath.replace(/\//g, ".");
-      let stack = `${property} ${message}`.trim();
+
+      let stack: string | undefined = undefined;
+
       if ("missingProperty" in params) {
         property = property
           ? `${property}.${params.missingProperty}`
           : params.missingProperty;
+
+        const currentProperty = property.split(".").slice(-1)[0];
+
+        const parentSchemaTitle =
+          parentSchema?.[PROPERTIES_KEY]?.[currentProperty]?.title;
+
+        if (parentSchemaTitle) {
+          message = message?.replace(currentProperty, parentSchemaTitle);
+        }
+
         stack = message!;
+      } else {
+        let title = property.split(".").pop() as string;
```

Comment:
I find it interesting that you needed to do a cast here since property is a string, split should create an array of strings, so popping should still be a string.

Reply:
@heath-freenome if you look at the types for `pop`, it returns `T | undefined`

```ts
    /**
     * Removes the last element from an array and returns it.
     * If the array is empty, undefined is returned and the array is not modified.
     */
    pop(): T | undefined;
```

I would not call it a `cast` as nothing happens at runtime, this is purely at the type level, and I think a [type assertion](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#type-assertions) is the right call here.

Reply:
Ah, thanks for clarifying!